DOC: quantile: correct/simplify documentation

mdhaber opened the pull request:
gh-24588 noted inaccuracies in the `quantile` documentation. This PR corrects these and simplifies the descriptions of the various methods. It also refers users of `percentile`, `nanquantile`, `nanpercentile` to the documentation of `quantile` to reduce redunandancy and ensure that the notes of these functions do not get out of sync.

Closes gh-24588

@lorentzenchr

eendebakpt commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4067,144 +4067,9 @@ def percentile(a,
-    :math:`P(Y \\leq t) = \\frac{1}{\\sum_i w_i} \\sum_i w_i 1_{x_i \\leq t}`.
-    Only ``method="inverted_cdf"`` supports weights.
-
+    The behavior of `numpy.percentile` with percentage `q` is
```
The `np.quantile` and `np.percentile` (and also the `np.nanpercentile`, `np.nanquantile`) share the same code to validate the `weights`. That could be factored out to make clear in the code that `numpy.percentile` is exactly the same as `np.quantile` with `q/100`.

mdhaber replied:
I agree, but that is out of scope of this PR.

eendebakpt commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4417,7 +4282,10 @@ def quantile(a,
     Notes
     -----
-    In general, the quantile at probability level :math:`q` of a cumulative
+    Given a sample `a` from an underlying distribution, `quantile` provides a
```
Don't we require many samples from the underlying distribution? For me `a sample` reads as a single value (usually a list is provided as the first input parameter of quantile)

mdhaber replied:
People use "sample" to mean both a single observation and a set of observations. Specifically to avoid ambiguity, I reserve the word "observation" for a single observation and "sample" for a set of observations.

eendebakpt commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4429,126 +4297,65 @@ def quantile(a,
+    =============================== =============== ===============
+
+    Note that indices ``j`` and ``j + 1`` are clipped to the range ``0`` to
+    ``n-1`` when the results of the formula would be beyond the allowed range
```
```suggestion
    ``n - 1`` when the results of the formula would be outside the allowed range
```

mdhaber replied:
The first of these changes can be considered to be an objective correction because it would be more consistent with the previous line.

What is the rationale for the second? (Maybe to avoid repetition of the syllable "be"?)

eendebakpt replied:
For me "beyond" covers the case where `j > n - 1`, and "outside" covers the case where `j < 0` or `j > n - 1`. (disclaimer: I am not a native English speaker)

mdhaber replied:
I'm not sure that this usage is inconsistent with the meaning of "beyond", but I agree with your assessment of the connotations/typical usage, so "outside" may be the better choice. I'll change it.

mdhaber commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4429,126 +4432,63 @@ def quantile(a,
+    | ``weibull``                  | 6             | ``q``         |
+    | ``linear`` (default)         | 7             | ``1 - q``     |
+    | ``median_unbiased``          | 8             | ``q/3 + 1/3`` |
+    | ``normal_unbiased``          | 9             | ``q/4 + 3/8`` |
```
```suggestion
    +-------------------------------+---------------+---------------+
    | ``method``                    | # in H&F [1]_ | ``m``         |
    +===============================+===============+===============+
    | ``interpolated_inverted_cdf`` | 4             | ``0``         |
    | ``hazen``                     | 5             | ``1/2``       |
    | ``weibull``                   | 6             | ``q``         |
    | ``linear`` (default)          | 7             | ``1 - q``     |
    | ``median_unbiased``           | 8             | ``q/3 + 1/3`` |
    | ``normal_unbiased``           | 9             | ``q/4 + 3/8`` |
    +-------------------------------+---------------+---------------+
```

mdhaber commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4429,126 +4297,65 @@ def quantile(a,
-    midpoint:
-        NumPy method kept for backwards compatibility.
-        Uses ``(i + j) / 2``.
+    =============================== =============== ===============
+    ``method``                      # in H&F [1]_   ``m``
+    =============================== =============== ===============
+    ``interpolated_inverted_cdf``   4               ``0``
+    ``hazen``                       5               ``1/2``
+    ``weibull``                     6               ``q``
+    ``linear`` (default)            7               ``1 - q``
+    ``median_unbiased``             8               ``q/3 + 1/3``
+    ``normal_unbiased``             9               ``q/4 + 3/8``
+    =============================== =============== ===============
```
```suggestion
    +-------------------------------+---------------+---------------+
    | ``method``                    | # in H&F [1]_ | ``m``         |
    +===============================+===============+===============+
    | ``interpolated_inverted_cdf`` | 4             | ``0``         |
    | ``hazen``                     | 5             | ``1/2``       |
    | ``weibull``                   | 6             | ``q``         |
    | ``linear`` (default)          | 7             | ``1 - q``     |
    | ``median_unbiased``           | 8             | ``q/3 + 1/3`` |
    | ``normal_unbiased``           | 9             | ``q/4 + 3/8`` |
    +-------------------------------+---------------+---------------+
```

mdhaber commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4429,126 +4297,65 @@ def quantile(a,
+    +-------------------------------+---------------+---------------+
+
+    Note that indices ``j`` and ``j + 1`` are clipped to the range ``0`` to
+    ``n - 1`` when the results of the formula would be outside the allowed range
+    of non-negative indices.
```
```suggestion
    ``n - 1`` when the results of the formula would be outside the allowed
    range of non-negative indices.
```

mdhaber commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4429,126 +4297,65 @@ def quantile(a,
-    midpoint:
-        NumPy method kept for backwards compatibility.
-        Uses ``(i + j) / 2``.
+    +-------------------------------+---------------+---------------+
+    | ``method``                    | # in H&F [1]_ | ``m``         |
+    +===============================+===============+===============+
+    | ``interpolated_inverted_cdf`` | 4             | ``0``         |
+    | ``hazen``                     | 5             | ``1/2``       |
+    | ``weibull``                   | 6             | ``q``         |
+    | ``linear`` (default)          | 7             | ``1 - q``     |
+    | ``median_unbiased``           | 8             | ``q/3 + 1/3`` |
+    | ``normal_unbiased``           | 9             | ``q/4 + 3/8`` |
+    +-------------------------------+---------------+---------------+
```
This table is not showing up in the rendered output. Any suggestions?

seberg replied:
```
/home/circleci/repo/venv/lib/python3.11/site-packages/numpy/__init__.py:docstring of numpy.quantile:144: ERROR: Malformed table.

+-------------------------------+---------------+---------------+
| ``method``                    | # in H&F [Re01cd3f3acfe-1]_ | ``m``         |
+===============================+===============+===============+
| ``interpolated_inverted_cdf`` | 4             | ``0``         |
| ``hazen``                     | 5             | ``1/2``       |
| ``weibull``                   | 6             | ``q``         |
| ``linear`` (default)          | 7             | ``1 - q``     |
| ``median_unbiased``           | 8             | ``q/3 + 1/3`` |
| ``normal_unbiased``           | 9             | ``q/4 + 3/8`` |
+-------------------------------+---------------+---------------+
```
Is it the reference being auto-expanded by numpy-doc maybe?  Or does sphinx not like the `=`?  Not sure.

mdhaber replied:
I'll try taking out the reference. That's probably it. Thanks! I saw that CircleCI was green and didn't look closer.

mdhaber replied:
Hmm that warning is gone but it's still not rendering appropriately.

<img width="582" alt="image" src="https://github.com/numpy/numpy/assets/6570539/c6a0e67a-2dde-47f3-a48d-87cd7a4803cd">

seberg replied:
I saw you had a try with using `===== ===== ====` style table?  That seeme dlike a good idea to me:  It is what the sort kind table uses as well.

mdhaber commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4429,126 +4297,65 @@ def quantile(a,
-        NumPy method kept for backwards compatibility.
-        Uses ``(i + j) / 2``.
+    +-------------------------------+---------------+---------------+
+    | ``method``                    | # in H&F [1]_ | ``m``         |
```
```suggestion
    | ``method``                    | # in H&F      | ``m``         |
```

mdhaber commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4429,126 +4297,65 @@ def quantile(a,
-    midpoint:
-        NumPy method kept for backwards compatibility.
-        Uses ``(i + j) / 2``.
+    +-------------------------------+---------------+---------------+
+    | ``method``                    | # in H&F      | ``m``         |
+    +===============================+===============+===============+
+    | ``interpolated_inverted_cdf`` | 4             | ``0``         |
+    | ``hazen``                     | 5             | ``1/2``       |
+    | ``weibull``                   | 6             | ``q``         |
+    | ``linear`` (default)          | 7             | ``1 - q``     |
+    | ``median_unbiased``           | 8             | ``q/3 + 1/3`` |
+    | ``normal_unbiased``           | 9             | ``q/4 + 3/8`` |
+    +-------------------------------+---------------+---------------+
```
```suggestion
     =============================== ========== =============== 
       ``method``                      # in H&F   ``m``          
      =============================== ========== =============== 
       ``interpolated_inverted_cdf``   4          ``0``          
       ``hazen``                       5          ``1/2``        
       ``weibull``                     6          ``q``          
       ``linear`` (default)            7          ``1 - q``      
       ``median_unbiased``             8          ``q/3 + 1/3``  
       ``normal_unbiased``             9          ``q/4 + 3/8``  
      =============================== ========== =============== 
```

mdhaber replied:
```suggestion
    =============================== =============== ===============
    ``method``                      # in H&F        ``m``
    =============================== =============== ===============
    ``interpolated_inverted_cdf``   4               ``0``
    ``hazen``                       5               ``1/2``
    ``weibull``                     6               ``q``
    ``linear`` (default)            7               ``1 - q``
    ``median_unbiased``             8               ``q/3 + 1/3``
    ``normal_unbiased``             9               ``q/4 + 3/8``
    =============================== =============== ===============
```

mdhaber commented:
Finally.

<img width="693" alt="image" src="https://github.com/numpy/numpy/assets/6570539/0eca1716-54be-4f95-9b7f-16d1c2249e5a">

I guess it needed to be the simplified table markup and not include a reference link. @lorentzenchr @melissawm I think this is ready for a look when you have a moment.

rossbar commented:
> and not include a reference link.

Sorry for catching this late - for future reference some numpydoc features don't work inside tables - there is a warning to this effect [buried deep in the docs](https://numpydoc.readthedocs.io/en/latest/format.html#references)

lorentzenchr commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4429,126 +4297,65 @@ def quantile(a,
+    empirical distribution function of the data, i.e.
+    :math:`F(y) = \\frac{1}{n} \\sum_i 1_{a_i \\leq y}`.
+    Then, different methods correspond to different choices of :math:`x` that
+    fulfill the above inequalities.
```
```suggestion
    fulfill the above coverage conditions.
```

lorentzenchr reviewed (commented):
This PR makes the reading much easier.
Note that the implementation follows the old text with "virtual indices". Now, the code gets even harder to read. I don't know how to best improve that.

mdhaber commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4429,126 +4297,65 @@ def quantile(a,
-
-        * if g > 0 ; then take j
-        * if g = 0 ; then average between bounds
+    By default (``method='linear'``), this is done by interpolating between
```
One of the recommendations in gh-24588 - and what I was inclined to do - was to start by addressing the default method before diving into the general framework. This lowers the initial complexity slightly, and I suspect that the majority of users are only interested in the default anyway. I've struck a compromise here: even though it starts as though it's only describing the default method, the transition to the general case is a very small step. I think this worked out nicely, so I'd prefer to keep it for now. We can leave the suggestion open so the maintainer who will review this (maybe @melissawm?) can see the suggestion.

melissawm replied:
Maybe something like:

```suggestion
    The optional method parameter specifies the method to use when the desired quantile is non-integer. By default (``method='linear'``), this is done by interpolating between
```

I use "non-integer" as a way to express that the result is somehow ambiguous and an arbitrary decision must be made. Feel free to reword it, but the gist of this suggestion is mentioning that this decision/ambiguity is the reason for the ``method`` parameter.

mdhaber replied:
To address this, I've removed the mention of the `method` parameter until later. Now we start with the simple default, then (only for readers who are still interested) introduce the `method` parameter.

mdhaber commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4429,126 +4297,65 @@ def quantile(a,
-        * if g = 0 and index is odd ; then take j
-        * if g = 0 and index is even ; then take i
+    where the index ``j`` and coefficient ``g`` are the integral and
+    fractional components of ``q * (n-1)``, and ``n`` is the number of
```
This would be `q * (n-1) + 1` in a 1-indexed system. However, when we get down to the general method, there is a `- 1` that can be attributed to Python being 0-indexed. I can mention it there.

lorentzenchr replied:
Right, got confused.

mdhaber commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4429,126 +4297,65 @@ def quantile(a,
-    midpoint:
-        NumPy method kept for backwards compatibility.
-        Uses ``(i + j) / 2``.
+    =============================== =============== ===============
```
This is another thing for the reviewing maintainer to consider. I would agree if the relationship between all the methods were simpler. Unfortunately, methods 1-3 use three different definitions of `g`, which breaks the nice interpretation of `j` and `g` being the "integer and fractional components" of the same quantity. We could add a column to the table, but I left them out intentionally since these methods are quite different.

melissawm replied:
Again, the compromise would be to mention this somehow, as a note to the reader that this table is not complete. Maybe something like move one sentence from a couple paragraphs down and write:

```
For the continuous estimators defined in [1]_, the table below shows how ``m`` is computed as a function of probability `q`:
```

mdhaber replied:
There are only three sentences that separate this table of continuous estimators from the discontinuous estimators below. 

The first is a minor footnote about clipping the indices. If we cannot make this an actual footnote, the best place for it is immediately after the table. (Before the table would interrupt the explanation, and a Note admonition would only increase the separation between the table and list.)

The next two explain that the table includes only the continuous estimators and introduces the discontinuous estimators from that same paper. I've modified it slightly (below) to emphasize that the table does not include all the estimators from that paper.

mdhaber commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4429,126 +4297,65 @@ def quantile(a,
-    interpolated_inverted_cdf:
-        method 4 of H&F [1]_.
-        This method gives continuous results using:
+    This a special case of Equation 1 of H&F [1]_. More generally,
```
Will do.

mdhaber reviewed (commented):
Thanks for the review. I'll make changes after other comments have come in.

lorentzenchr commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4417,7 +4282,10 @@ def quantile(a,
+    More formally, the quantile at probability level :math:`q` of a cumulative
     distribution function :math:`F(y)=P(Y \\leq y)` with probability measure
     :math:`P` is defined as any number :math:`x` that fulfills the
     *coverage conditions*
```
I would definitely keep the coverage conditions. They are kind of the defining equations for a quantile in statistical terms. As estimations, sample quantiles are allowed to deviate. Note that P2 of HF is one of the coverage conditions.

mdhaber replied:
Yes, I've kept them, and I've made your argument contiguous again.

melissawm reviewed (commented):
I'm sorry I dropped the ball on this! I left a couple of comments but please keep in mind I'm no expert in this area.

One thing that stood out to me reading the full docstring is the mention of "R types" in the method parameter. A quick scan of the H&F paper didn't help me - I tried tracking this change in the docs but didn't really manage to find why this comment was added (it was added in 1.22.0). If you know what this means, is there a way we can explain this comment? Otherwise it can be confusing for readers.

melissawm commented:
Overall, much much better! Thank you so much @mdhaber and @lorentzenchr for the review!

mdhaber commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4429,126 +4297,65 @@ def quantile(a,
+    ``n - 1`` when the results of the formula would be outside the allowed
+    range of non-negative indices.
+
+    The methods above are all continuous functions of probability `q`.
+    Methods 1-3 of H&F [1]_ provide three discontinuous estimators, where
+    ``j`` is defined as above and ``m`` and ``g`` are defined as follows.
```
```suggestion
    The table above includes only the estimators from H&F that are continuous
    functions of probability `q` (estimators 4-9). NumPy also provides the
    three discontinuous estimators from H&F (estimators 1-3), where ``j`` is
    defined as above and ``m`` and ``g`` are defined as follows.
```

mdhaber commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4429,126 +4297,65 @@ def quantile(a,
-    nearest:
-        NumPy method kept for backwards compatibility.
-        Takes ``i`` or ``j``, whichever is nearest.
+    where ``m`` is defined differently for each value of the ``method``
+    parameter.
```
```suggestion
    where ``m`` may be defined according to several different conventions.
    The preferred convention may be selected using the ``method`` parameter:
```

mdhaber commented:
Thanks @melissawm. I think the "R types" refer to the "Types" listed in the documentation of [R's `quantile`](https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/quantile) function. I just removed this because it is confusing, and anyone already familiar with the R types will recognize these as being identical. I've also implemented changes to respond to the other comments.

melissawm reviewed (approved):
This looks great to me, thank you @mdhaber !

melissawm commented:
I'll leave it open in case @lorentzenchr wants to take a last look. Cheers!

mdhaber commented:
Since this has been close to its present state for a few months, I'd ask whether we can merge as-is, and if there are additional changes to be made, they can be made in a new PR?

lorentzenchr commented:
I'll have a quick look...

lorentzenchr commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4400,139 +4267,82 @@ def quantile(a,
-    distribution function :math:`F(y)=P(Y \\leq y)` with probability measure
-    :math:`P` is defined as any number :math:`x` that fulfills the
-    *coverage conditions*
+    Given a sample `a` from an underlying distribution, `quantile` provides a
```
Sometimes it's `` `a` `` sometimes its ``` ``a`` ``` and it renders differently.

mdhaber replied:
According to numpydoc standards , single backticks are correct here because `a` is an argument of this function. I see that you added some double backticks in gh-24254 that I didn't notice, since I wrote this before that PR merged. I'll fix that.

lorentzenchr commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4400,139 +4267,82 @@ def quantile(a,
-
-    .. math::
-        i + g = q * ( n - alpha - beta + 1 ) + alpha
+    By default, this is done by interpolating between adjacent elements in
```
```suggestion
    By default (`method="linear"`), this is done by interpolating between adjacent elements in
```

mdhaber replied:
I had this, but removed it to respond to another comment.

lorentzenchr commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4400,139 +4267,82 @@ def quantile(a,
-        method 2 of H&F [1]_.
-        This method gives discontinuous results:
+    - ``j = (q*n + m - 1) // 1``, and
+    - ``g = (q*n + m - 1) % 1``,
```
Maybe add here, that the `-1` is for translating from 1-based (H&F) to 0-based (Python) indexing.

mdhaber replied:
Yes, this is the place I mentioned [here](https://github.com/numpy/numpy/pull/25704#discussion_r1477787361) where the `-1` can be attributed to the 0-based indexing. I suppose I point this out.

lorentzenchr commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4400,139 +4267,82 @@ def quantile(a,
 
-        * alpha = 0
-        * beta = 1
+    1. ``inverted_cdf``: ``m = 0`` and ``g = int(q*n > 0)``
```
In my experience, this is the version used the most in statistical literature and corresponds to H&F's first equation in the introduction. But this equation is not general enough, it neglects distributions with flat regions, e.g. step functions.

Summary: better no action.

mdhaber replied:
I don't understand. Does "better no action" mean that there is no action to take to respond to this comment, or that I should not have taken action to adjust this part of the documentation?

lorentzenchr replied:
Yes, I suggest to leave it as is. Just wanted to share thoughts with a fellow statistian.

lorentzenchr commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4400,139 +4267,82 @@ def quantile(a,
-        NumPy method kept for backwards compatibility.
-        Uses ``(i + j) / 2``.
+    with random variable :math:`Y\\sim P`.
+    Some of the quantile estimators above (namely ``inverted_cdf`` and
+    ``averaged_inverted_cdf``) arise when one considers :math:`F` as the
```
```suggestion
    Some of the quantile estimators above, namely ``inverted_cdf`` and
    ``averaged_inverted_cdf``, arise when one considers :math:`F` as the
```

mdhaber replied:
The most objective correction here would be to add a comma _after_ "namely". Use of parentheses or commas to surround such a phrase are both valid, as far as I can tell (e.g., https://english.stackexchange.com/questions/145452/punctuation-around-the-word-namely).

lorentzenchr replied:
I suggested it, on particular because I thought you don’t like parentheses in a sentence.

mdhaber replied:
https://github.com/numpy/numpy/pull/25704/files#r1590014290

mdhaber replied:
```suggestion
    Some estimators arise when one considers :math:`F` as the empirical
```

lorentzenchr reviewed (commented):
Minor comments. Would be good without them.
I really like this doc improvement. It reads so much better!

I have no rights to approve, but consider it a LGTM.

mdhaber commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4400,139 +4267,82 @@ def quantile(a,
 
-        * if g > 0 ; then take j
-        * if g = 0 ; then average between bounds
+    where ``m`` may be defined according to several different conventions.
```
To respond to [this](https://github.com/numpy/numpy/pull/25704/files#r1589734258),
```suggestion
    where the ``- 1`` in these formulas accounts for Python's 0-based indexing,
    and ``m`` may be defined according to several different conventions.
```
I suppose I left this out original because I chose not to explain all of the ways in which the formulas are not written as they are in the paper. For instance, I also use `%` to take the fractional part rather than subtracting the integral part. This seems equally notable to me. While adding such information would help those who want to compare this to the original text, I think make it harder to follow for those who are reading only this document.

lorentzenchr replied:
`//` and `%` are clear and don’t need explanation.
The different base of indexing is in my opinion worth mentioning somewhere. It took me a while to figure it out.
Maybe a good place is below the reference.

mdhaber replied:
https://github.com/numpy/numpy/pull/25704/files#r1590013796

mdhaber commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4400,139 +4267,82 @@ def quantile(a,
-    with random variable :math:`Y\\sim P`.
     Sample quantiles, the result of ``quantile``, provide nonparametric
     estimation of the underlying population counterparts, represented by the
     unknown :math:`F`, given a data vector ``a`` of length ``n``.
```
```suggestion
    unknown :math:`F`, given a data vector `a` of length ``n``.
```

mdhaber commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4400,139 +4267,82 @@ def quantile(a,
-        * if g = 0 and index is even ; then take i
+    Note that indices ``j`` and ``j + 1`` are clipped to the range ``0`` to
+    ``n - 1`` when the results of the formula would be outside the allowed
+    range of non-negative indices.
```
```suggestion
    range of non-negative indices. The ``- 1`` in the formulas for ``j`` and
    ``g`` accounts for Python's 0-based indexing.
```

mdhaber commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4400,139 +4267,82 @@ def quantile(a,
+    empirical distribution function of the data, i.e.
+    :math:`F(y) = \\frac{1}{n} \\sum_i 1_{a_i \\leq y}`.
+    Then, different methods correspond to different choices of :math:`x` that
+    fulfill the above coverage conditions.
```
```suggestion
    fulfill the above coverage conditions. Methods that follow this approach are
    ``inverted_cdf`` and ``averaged_inverted_cdf``.
```

lorentzenchr commented:
@seberg Why did you close?

lorentzenchr commented on numpy/lib/_function_base_impl.py:
```diff
@@ -4400,139 +4267,83 @@ def quantile(a,
+    empirical distribution function of the data, i.e.
+    :math:`F(y) = \\frac{1}{n} \\sum_i 1_{a_i \\leq y}`.
+    Then, different methods correspond to different choices of :math:`x` that
+    fulfill the above coverage conditions. Methods that follow this approach are
+    ``inverted_cdf`` and ``averaged_inverted_cdf``.
```
```suggestion
    fulfill the above coverage conditions. Methods that follow this approach
    are ``inverted_cdf`` and ``averaged_inverted_cdf``.
```
line too long

seberg commented:
@seberg Why did you close?

Ouch, I honstly don't know how that happened!

mdhaber committed:
DOC: quantile: correct/simplify documentation

seberg commented:
Had a quick read through and it looks good and it I think things were settling.  Thanks everyone, let's get this in!
